services/sendgrid/v2/emailService.go

```diff
@@ -112,5 +112,29 @@ func (s *sendgridEmailService) SendEmailVerificationEmail(ctx context.Context, u
 }
 
 func (s *sendgridEmailService) SendPasswordResetEmail(ctx context.Context, user entities.User, passwordResetResources common.UniformResourceIdentifiers) error {
-	panic("not implemented")
+	emailToken, err := s.authorizer.CreateServiceToken(ctx, user.ID,
+		passwordResetResources, s.timeProvider.Now().Unix()+s.cfg.Email.TokenLifetime)
+	if err != nil {
+		return errors.Wrap(err, "could not create auth token for email")
+	}
+
+	resetURL := fmt.Sprintf("http://%s/resetpwd?token=%s", s.cfg.AppURL, emailToken)
+
+	var contentBuff bytes.Buffer
+	err = s.passwordResetEmailTemplate.Execute(&contentBuff, emailTemplateDataModel{
+		Link:       resetURL,
+		SenderName: s.cfg.Email.NoreplyEmailName,
+	})
+	if err != nil {
+		return errors.Wrap(err, "could not construct email")
+	}
+
+	return s.SendEmail(
+		s.cfg.Email.PasswordResetEmailSubj,
+		contentBuff.String(),
+		contentBuff.String(),
```

Comment:
Yeah that's probably not correct. I copied this function from EmailService v1 so I guess it was fine before but we should definitely fix this at some point. I'll add a TODO because I'd rather not spend more time on fixing this service client since we'll be replacing it quite soon anyway.

Reply:
https://github.com/unicsmcr/hs_auth/issues/120